browser_history/generic.py

```diff
@@ -175,56 +189,95 @@ def fetch(self, history_paths=None, sort=True, desc=False):
+        :type bookmarks_paths: list(:py:class:`pathlib.Path`)
+        :param sort: (optional) flag to specify if the output should be sorted.
+            Default value set to True.
+        :type sort: boolean
+        :param desc: (optional)  flag to speicify asc/desc (Applicable iff sort is True)
+            Default value set to False.
+        :type asc: boolean
+        :return: Object of class :py:class:`browser_history.generic.Outputs` with the
+            data member entries set to list(tuple(:py:class:`datetime.datetime`, str))
+        :rtype: :py:class:`browser_history.generic.Outputs`
+        """
+
+        if bookmarks_paths is None:
+            bookmarks_paths = self.paths(profile_file=self.bookmarks_file)
+        output_object = Outputs(fetch_type="bookmarks")
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            for bookmarks_path in bookmarks_paths:
+                if not os.path.exists(bookmarks_path):
+                    continue
+                copied_bookmark_path = shutil.copy2(
+                    bookmarks_path.absolute(), tmpdirname
+                )
+                date_bookmarks = self.bookmarks_parser(copied_bookmark_path)
+                output_object.bookmarks.extend(date_bookmarks)
+                if sort:
+                    output_object.bookmarks.sort(reverse=desc)
         return output_object
 
 
 class Outputs:
     """
-    A generic class to encapsulate history outputs and to
+    A generic class to encapsulate history and bookmark outputs and to
     easily convert them to JSON, CSV or other formats.
 
-    * **entries**: List of tuples of Timestamp & URL
-    :type entries: list(tuple(:py:class:`datetime.datetime`, str))
+    * **histories**: List of tuples of Timestamp & URL
+    :type history_entries: list(tuple(:py:class:`datetime.datetime`, str))
 
-    * **formats**: A tuple of strings containing all supported formats
+    * **bookmarks** List of tuples of Timestamp , URL , Title , Folder
+    :type bookmark_entries: list(tuple(:py:class:`datetime.datetime`, str,str,str))
 
-    * **fields**: The fields available for the history data returned
+    * **formats**: A tuple of strings containing all supported formats
 
     """
 
     # All formats added here should be implemented in _format_map
     # Formats added here and in _format_map should be in lowercase
     formats = ("csv", "json")
-  
+
     # Use the below fields for all formatter implementations
-    fields = ("Timestamp", "URL")
 
-    def __init__(self):
-        self.entries = []
+    def __init__(self, fetch_type):
+        self.histories = []
+        self.bookmarks = []
         # format map is used by the formatted method to call the right formatter
         self._format_map = {
             "csv": self.to_csv,
             "json": self.to_json,
-            "jsonl": lambda: self.to_json(json_lines=True)
+            "jsonl": lambda: self.to_json(json_lines=True),
         }
```

Comment:
You could make this a class variable (outside init) since it will be the same for all objects.

Reply:
okay

Reply:
This might not be possible as it has to access the self variable and it throws a error if its kept outside

Reply:
Ah okay. Then I guess you can just add `jsonl` to `formats`.

But we need a way to ensure that `formats` has all supported formats and that all supported formats are listed in `formats`.

You could write a simple test for this - to ensure that `formats` has the same elements as `_format_map.keys()`.

Reply:
We can still access the elements through format_map,keys(), I just told if we kept it outside the __init__ function then it causes the error, I can still do `Outputs(fetch_type =None).format_map.keys()`

Reply:
You cannot instantiate the class in the definition itself, right?
`formats` will need `Outputs` to be defined which needs `formats` which needs `Outputs`... you see where this is going xD

Reply:
this goes in the cli.py though not inside the class Outputs, shouldnt that work?
I still mean keeping self.format_map inside the __init__function itself, if it needs to be accessed inside the Outputs function the self variable is used

Reply:
I added a commit , thats what I did

Reply:
Ah okay. Makes sense. I was thinking of the `formats` variable inside `Outputs` class. Your solution is nice :+1: